themes/default/content/blog/kubernetes-server-side-apply/index.md

```diff
@@ -0,0 +1,738 @@
+---
+title: "Manage Shared Kubernetes Resources Safely with Pulumi"
+
+date: 2022-07-19T17:13:34-06:00
+
+meta_desc: Manage shared Kubernetes resources safely with our new Server-Side Apply support
+meta_image: meta.png
+authors:
+    - levi-blackstone
+tags:
+    - kubernetes
+---
+
+Kubernetes resources often have more than one controller making changes to them. These controllers can include `kubectl`, the Kubernetes control plane, custom operators, or infrastructure as code (IaC) tools like Pulumi.
+With the [v3.20.1 release](https://github.com/pulumi/pulumi-kubernetes/releases/tag/v3.20.1) of the Kubernetes provider, you have some powerful new options for managing shared resources in Kubernetes. In this post, we show you
+how Pulumi can help you work with shared resources safely and effectively.
+
+<!--more-->
+
+## Adding labels to an existing Namespace
+
+Some Kubernetes resources, like Namespaces, are shared by all users of the cluster. Users often want to edit these resources, perhaps by adding a new label to them. In this example, we add a label to the `kube-public` Namespace using the new `NamespacePatch` resource:
+
+{{% notes type="info" %}}
+Multiple Patch resources can modify the same Kubernetes resource.
+{{% /notes %}}
+
+{{< chooser language "typescript,python,go,csharp,java,yaml" >}}
+
+{{% choosable language typescript %}}
+
+```typescript
+import * as pulumi from "@pulumi/pulumi";
+import * as kubernetes from "@pulumi/kubernetes";
+
+const provider = new kubernetes.Provider("provider", {enableServerSideApply: true});
```

Comment:
Do we need to call out the magic sauce is `enableServiceSideApply` in our provider? Can we also set this any other way if there's no explicit provider in play today?

Reply:
I cover this in more detail in the how-to guide linked in the last section, but perhaps I should add another link to it up front?

Reply:
Added a summary and link to the how-to guide in the first section.